vnext/Microsoft.ReactNative/IReactNotificationService.h

```diff
@@ -30,6 +38,30 @@ struct ReactNotificationArgs : implements<ReactNotificationArgs, IReactNotificat
   IInspectable m_data;
 };
 
+// The ReactNotificationService manages a list subscriptions.
+// Subscriptions can be added or removed using Subscribe, Unsubscribe, and UnsubscribeAll methods.
+// The SendNotification sends notifications to all subscription in the list.
+// When ReactNotificationService is destroyed, it calls the UnsubscribeAll method.
+//
+// A notification service may have a "parent" notification service.
+// It allows a group of subscriptions to be automatically removed when the "child" service is destroyed.
+// We use this mechanism to automatically unsubscribe on React instance unload from all subscriptions
+// done in the native modules.
+// This is how the parent-child relationship works:
+// - All notifications are sent from the "root" service - the topmost "parent" service. SendNotification
+//   method called on the "child" service just calls "parent" service's SendNotification method.
+// - Subscriptions are added to services on all levels: a subscription added to a "child" service is also
+//   added to the "parent" service.
+// - When "child" subscription is removed, it also removes the corresponding subscription in the
```

Comment:
Do we have a risk of folks adding the same subscription to two nodes? i.e add it to root and then create a child to which it is added? 
Of with the use of Lambda's to register the notifications that risk does not really exist... Or does the c++ parameterless lambda do some shared instance management?

Reply:
Each subscription is an object and it is unique. The `Unsubscribe` method is only available on the `IReactNotificationSubscription` object, an not on the `IReactNotificationService`. I.e. every time we call `Subscribe` method we create a unique subscription even if some of them have the same notification Id and the callback/lambda.

Reply:
Sounds good, thanks!

---
In reply to: [557833370](https://github.com/microsoft/react-native-windows/pull/6877#discussion_r557833370) [](ancestors = 557833370)